athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -1031,6 +1053,271 @@ else if (value instanceof Boolean && (boolean) value) {
         }
     }
 
+    /**
+     * Used to write a Map value.
+     *
+     * @param allocator The BlockAllocator which can be used to generate Apache Arrow Buffers for types
+     * which require conversion to an Arrow Buffer before they can be written using the FieldWriter.
+     * @param writer The FieldWriter for the Map field we'd like to write into.
+     * @param field The Schema details of the Map Field we are writing into.
+     * @param pos The position (row) in the Apache Arrow batch we are writing to.
+     * @param value The value we'd like to write as a Map.
+     * @param resolver The field resolver that can be used to extract individual Struct Map from the value.
+     */
+    @VisibleForTesting
+    protected static void writeMap(BufferAllocator allocator,
+                                      UnionMapWriter writer,
+                                      Field field,
+                                      int pos,
+                                      Object value,
+                                      FieldResolver resolver)
+    {
+        //We expect null writes to have been handled earlier so this is a no-op.
+        if (value == null) {
+            return;
+        }
+
+        //Indicate the beginning of the Map value, this is how Apache Arrow handles the variable length of map types.
+        writer.startMap();
+
+        List<Field> children = field.getChildren();
+        Field keyValueStructField;
+        if (children.size() != 1) {
+            throw new IllegalStateException("Invalid Arrow Map schema: " + field);
+        }
+        else {
+            keyValueStructField = children.get(0);
+            if (!ENTRIES.equals(keyValueStructField.getName()) || !(keyValueStructField.getType() instanceof ArrowType.Struct)) {
+                throw new IllegalStateException("Invalid Arrow Map schema: " + field);
+            }
+        }
+
+        List<Field> keyValueChildren = keyValueStructField.getChildren();
+        Field keyField;
+        Field valueField;
+        if (keyValueChildren.size() != 2) {
+            throw new IllegalStateException("Invalid Arrow Map schema: " + field);
+        }
+        else {
+            keyField = keyValueChildren.get(0);
+            valueField = keyValueChildren.get(1);
+            if (!KEY.equals(keyField.getName()) || !VALUE.equals(valueField.getName())) {
+                throw new IllegalStateException("Invalid Arrow Map schema: " + field);
+            }
+        }
+
+        for (Field nextChild : keyValueChildren) {
+            //For each child field that comprises the Map, attempt to extract and write the corresponding value
+            //using the FieldResolver.
+            Object childValue = resolver.getFieldValue(nextChild, value);
+            switch (Types.getMinorTypeForArrowType(nextChild.getType())) {
```

Comment:
Do we need to support nested maps?

Reply:
not at the moment, this is for basic maps. We can enhance later on.

Reply:
We should. Maps with struct or list values are a valid case.